internal/core/writer/properties/client.go

```diff
@@ -195,6 +225,41 @@ func (dpc *DimensionPropertyClient) isDuplicate(dimProps *types.DimProperties) b
 	return ok && reflect.DeepEqual(prev.(*types.DimProperties), dimProps)
 }
 
+func (dpc *DimensionPropertyClient) fetchExistingDimension(key, value string) (map[string]string, []string, error) {
+	url, err := dpc.makeDimURL(key, value)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	resp, err := dpc.client.Get(url.String())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if resp.StatusCode != 200 {
+		// No existing dimension value, so return blank for both.
```

Comment:
Checking `!= 200` for this is dangerous. No dimension object would be a 404; you may get other status codes when throttled, or even a 502 or 503 if SignalFx as a partial outage. In which case you don't want to end up sending a PUT with none of the original metadata.

Reply:
Ok, made it so that only a 404 returns the empty property/tag set.  Anything >=500 returns error, which now causes an indefinite retry with exponential backoff (had to refactor some things to make it not block other non-merging dim updates).  Am now basically assuming that anything not a 404 or 5xx response is "ok", although subsequent json parse errors will also trigger an error and thus the retry.